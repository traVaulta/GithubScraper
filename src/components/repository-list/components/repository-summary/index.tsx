import React, {useState} from 'react';
import {RepositorySummaryDTO} from '../../models';
import Emoji from "react-emoji-render";

export const RepositorySummary = (props: { repositorySummary: RepositorySummaryDTO }) => {
    const [expanded, toggle] = useState(false);
    return (
        <>
            <div className="panel panel--info" onClick={() => toggle(!expanded)}>
                <span><a href={props.repositorySummary.url}>{props.repositorySummary.name}</a></span>
            </div>
            {expanded &&
                <div className="panel panel--info description-text">
                    <Emoji text={props.repositorySummary.description || 'No description available...'} />
                </div>
            }
        </>
    );
};
