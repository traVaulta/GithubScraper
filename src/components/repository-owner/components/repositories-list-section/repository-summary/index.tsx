import React, {useState, MouseEvent} from 'react';
import {RepositorySummaryDTO} from '../../../models';
import Emoji from 'react-emoji-render';

import {noBubbling} from '../../../../shared/utils/event-handlers';

export const RepositorySummary = (props: { repositorySummary: RepositorySummaryDTO }) => {
    const [expanded, toggle] = useState(false);
    const toggleSummary = (_: MouseEvent) => toggle(!expanded);
    const openInNewTab = (_: MouseEvent) => window.open(props.repositorySummary.url, '_blank');
    return (
        <>
            <div className="panel panel--info panel--content" onClick={noBubbling(toggleSummary)}>
                <div className="panel panel--info panel--title">
                    <a href={props.repositorySummary.url}>{props.repositorySummary.name}</a>
                </div>
                <div className="panel panel--info panel--actions">
                    <button onClick={noBubbling(openInNewTab)}><i className="fa fa-external-link"/></button>
                </div>
            </div>
            {expanded && (
                <div className="panel panel--info panel--content description-text">
                    <Emoji text={props.repositorySummary.description || 'No description available...'}/>
                </div>
            )}
        </>
    );
};
