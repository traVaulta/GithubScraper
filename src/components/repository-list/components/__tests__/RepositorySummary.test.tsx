import React from 'react';
import {render} from '@testing-library/react';

import {RepositorySummary} from '../repository-summary';

describe(`${RepositorySummary.name}`, () => {
    it('should pass', () => {
        const repositoryDTO = {
            name: 'AwesomeReactRepo',
            url: 'https://github.com/react/master'
        };
        const {getByText} = render(<RepositorySummary repositorySummary={repositoryDTO}/>);
        const element = getByText(/react/i);
        expect(element).toBeInTheDocument();
    });
});
