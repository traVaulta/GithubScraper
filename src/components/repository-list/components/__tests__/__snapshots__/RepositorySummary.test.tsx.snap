// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RepositorySummary should display default note when no description on click 1`] = `
"Snapshot Diff:
- First value
+ Second value

@@ -19,6 +19,15 @@
            class=\\"fa fa-external-link\\"
          />
        </button>
      </div>
    </div>
+   <div
+     class=\\"panel panel--info panel--content description-text\\"
+   >
+     <span
+       class=\\"\\"
+     >
+       No description available...
+     </span>
+   </div>
  </DocumentFragment>"
`;

exports[`RepositorySummary should display description on click 1`] = `
"Snapshot Diff:
- First value
+ Second value

@@ -19,6 +19,20 @@
            class=\\"fa fa-external-link\\"
          />
        </button>
      </div>
    </div>
+   <div
+     class=\\"panel panel--info panel--content description-text\\"
+   >
+     <span
+       class=\\"\\"
+     >
+       Awesome React project 
+       <span
+         style=\\"width: 1em; height: 1em; margin: 0px .05em 0px .1em; vertical-align: -0.1em;\\"
+       >
+         ðŸ˜„
+       </span>
+     </span>
+   </div>
  </DocumentFragment>"
`;
